### PR TITLE
netlink: Device speed is shown as 65535

### PR DESCRIPTION
**Pull request by jonnary**
This patch fix device speed 65535 to "unknown"

### CODE REVIEW

File: topology/probes/netlink/netlink.go [hunk 1]
```diff
@@ -401,6 +401,8 @@ func (u *NetNsProbe) addLinkToTopology(link netlink.Link) {
 	if speed, err := u.ethtool.CmdGet(&ethtool.EthtoolCmd{}, attrs.Name); err == nil {
 		if speed != math.MaxUint32 {
 			metadata["Speed"] = int64(speed)
+		}else {
+			metadata["Speed"] = "Unknown"
```
Comment (nplanel) [inline topology/probes/netlink/netlink.go]: may not be a good idea to have a metadata of type int64 and string
I think the real fix would be to change the line to speed != math.MaxUint16   => need to double check ethtool package and ethtool API

Comment (jonnary) [reply topology/probes/netlink/netlink.go]: 1、ethtool result
[root@host_192_51_151_25 ~]# ethtool ens44f1
Settings for ens44f1:
Supported ports: [ FIBRE ]
Supported link modes: 1000baseT/Full
10000baseT/Full
Supported pause frame use: No
Supports auto-negotiation: Yes
Advertised link modes: 1000baseT/Full
10000baseT/Full
Advertised pause frame use: No
Advertised auto-negotiation: Yes
Speed: Unknown!
Duplex: Unknown! (255)
Port: Other
PHYAD: 0
Transceiver: external
Auto-negotiation: on
Supports Wake-on: umbg
Wake-on: g
Current message level: 0x00000007 (7)
drv probe link
Link detected: no
2、safchain/ethtool
pull a request: https://github.com/safchain/ethtool/pull/22
3、in skydive maybe we can changed as under?
                // TODO: it's better to fix math.MaxUint16 value in safchain/ethtool project
		if speed != math.MaxUint32 || speed != math.MaxUint16 {
			metadata["Speed"] = int64(speed)
		} else {
			metadata["Speed"] = "Unknown"
		}

Comment (safchain) [reply topology/probes/netlink/netlink.go]: Sorry I'm going to elaborate a bit. The datastore won't support multiple types for a same attribute, moreover we plan to convert metadata to protobuf which avoid multiple types as well. 

If the speed is unknown I propose to not report it.

**Comment by jonnary** [comment]
OK.
As https://github.com/safchain/ethtool/pull/22 has merged, the  fix in skydive can be as follow?
1.upgrade skydive/vendor.json
2.If the speed is unknown we choice not report it or we report it int64(-1)?

**Comment by safchain** [comment]
I would prefer to not report it

**Comment by jonnary** [comment]
OK:)

**Comment by lebauce** [comment]
Fixed by https://github.com/skydive-project/skydive/pull/1576

**Pull request closed without merging**